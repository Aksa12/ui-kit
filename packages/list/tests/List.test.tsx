import React from "react";
import { shallow } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";

import { List } from "../";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("List", () => {
  it("renders default", () => {
    const component = shallow(
      <List tag="ul">
        <li>list item</li>
        <li>list item</li>
      </List>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with all props", () => {
    const component = shallow(
      <List markerStyle="disc" tag="ol">
        <li>list item</li>
        <li>list item</li>
      </List>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
