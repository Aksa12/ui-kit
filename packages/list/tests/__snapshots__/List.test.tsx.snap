// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders default 1`] = `
.emotion-0 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  list-style-type: none;
}

<ul
  className="emotion-0"
  data-cy="list"
>
  <li>
    list item
  </li>
  <li>
    list item
  </li>
</ul>
`;

exports[`List renders with all props 1`] = `
.emotion-0 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  list-style-type: disc;
  padding-left: 16px;
}

<ol
  className="emotion-0"
  data-cy="list"
>
  <li>
    list item
  </li>
  <li>
    list item
  </li>
</ol>
`;
